src/ros/mb_publisher.cpp

```diff
@@ -18,6 +18,36 @@ void MBPublisher::publish(ur_msgs::IOStates& io_msg, SharedMasterBoardData& data
   io_pub_.publish(io_msg);
 }
 
+void MBPublisher::publishRobotStatus(SharedRobotModeData& data)
+{
+  industrial_msgs::RobotStatus msg;
+  msg.drives_powered.val = data.robot_power_on;
+  msg.e_stopped.val = data.emergency_stopped;
+  msg.in_motion.val = data.program_running;
+  msg.error_code = 0;
+  msg.in_error.val = data.protective_stopped;
+  msg.mode.val = industrial_msgs::TriState::UNKNOWN;
+
+  status_pub_.publish(msg);
+}
+
+void MBPublisher::publishRobotStatus(RobotModeData_V3_0__1& data)
+{
+  industrial_msgs::RobotStatus msg;
+
+  msg.drives_powered.val = data.robot_power_on;
+  msg.e_stopped.val = data.emergency_stopped;
+  msg.in_motion.val = data.program_running;
```

Comment:
I don't believe `in_motion` is the same as `program_running`, is it?

Reply:
Can secondary programs contain motion commands? And do they then update `program_running`?

Reply:
it is `if` but not `iff`, ie there will always be a program running when it moves but having a program running does not signify that it will actually move. At least that's my understanding

Reply:
Ok. But then this expression doesn't currently accurately encode the state of the robot, right?

It could be executing a program without any motion?

Reply:
`industrial_msgs::TriState::Unknown` then?

Reply:
we don't use this field except for decorative purposes, so I don't care from a consumer point of view

Reply:
> we don't use this field except for decorative purposes, so I don't care from a consumer point of view

That's great, but if we add this (`RobotStatus`) it should accurately encode the status of the robot.

If we can't derive something from what the robot provides us, then yes, I believe using `TriState::Unknown` for now would be better.